File: Review-12-Python3-Shapes/Example-5/shapes/Square.py
import copy

from shapes.Shape import (Shape, FPT_FMT)


class Square(Shape):
    """
    A Rectangle with 4 Equal Sides
    """

    def __init__(self, side=1):
        """
        Construct a Square
        """
        self.name = "Square"

        self._side = side

    @property
    def side(self):
        return self._side

    @side.setter
    def side(self, s):
        self._side = s

    def area(self):
        """
        Compute the area

        :return: area
        """
        return self._side ** 2.0

    def perimeter(self):
        """
        Compute the perimeter

        :return: perimeter
        """

        return 4 * self._side

    def __deepcopy__(self, memo):
        """
        Return a new duplicate Shape
        """

        return Square(copy.deepcopy(self.side))

    def __str__(self):
        """
        Print the Square
        """

        return (super(Square, self).__str__()
                + FPT_FMT.format("Side", self.side)
                + FPT_FMT.format("Perimeter", self.perimeter())
                + FPT_FMT.format("Area", self.area()))

File: Review-06-Java-Shapes/Example-8/src/main/java/edu/odu/cs/cs330/examples/shapes/Square.java
// Thomas Kennedy
// CS 330 Fall 2020

package edu.odu.cs.cs330.examples.shapes;

/**
 * A Rectangle with 4 Equal Sides.
 *
 * @author Thomas J Kennedy
 */
public class Square extends Shape implements Cloneable, TraitFromDimensions {
    /**
     * Length of One Side.
     */
    private double _side;

    /**
     * Construct a Square with side set to 1.
     */
    public Square()
    {
        this(1);
    }

    /**
     * Construct a Square.
     *
     * @param s the desired side length
     */
    public Square(double s)
    {
        _side = s;
    }

    /**
     * Construct a Square.
     *
     * @param src the Square to copy
     */
    public Square(Square src)
    {
        this(src._side);
    }

    /**
     * Return the side length.
     *
     * @return length of a single side
     */
    public double side()
    {
        return _side;
    }

    /**
     * Modify the side length.
     *
     * @param s the replacement length
     */
    public void side(double s)
    {
        _side = s;
    }

    @Override
    public String name()
    {
        return "Square";
    }

    @Override
    public double area()
    {
        return Math.pow(_side, 2.0);
    }

    @Override
    public double perimeter()
    {
        return 4 * _side;
    }

    @Override
    public Object clone() throws CloneNotSupportedException
    {
        return new Square(this);
    }

    @Override
    public int numDims()
    {
        return 1;
    }

    @Override
    public void createFromDims(double[] dims)
    {
        this._side = dims[0];
    }

    @Override
    public String toString()
    {
        return super.toString()
             + String.format(FMT_DBL, "Side", this._side)
             + String.format(FMT_DBL, "Perimeter", this.perimeter())
             + String.format(FMT_DBL, "Area", this.area());
    }
}





